src/tests/specialfunctions/test_specialfunctions_gamma.fypp

```diff
@@ -0,0 +1,587 @@
+
+        #:elif k1 == "int32"
+        
+        ${t1}$, parameter :: x(n) = [0_${k1}$, 1_${k1}$, 2_${k1}$, 4_${k1}$,   &
+                                     12_${k1}$, 7000_${k1}$]
+        real(sp), parameter :: ans(n) = [0.0, 0.0, 0.693147180, 3.17805383,    &
+                                         1.99872145e1, 5.49810038e4]
+
+        #:elif k1 == "int64"
+        
+        ${t1}$, parameter :: x(n) = [0_${k1}$, 1_${k1}$, 2_${k1}$, 4_${k1}$,   &
+                                     20_${k1}$, 90000_${k1}$]
+        real(sp), parameter :: ans(n) = [0.0, 0.0, 0.693147180, 3.17805383,    &
+                                         4.23356165e1, 9.36687468e5]
+
+        #:endif
+
+        do i = 1, n
+
+            call check(error, log_factorial(x(i)), ans(i), "Integer kind "     &
+                       //"${k1}$ failed", thr = tol_sp, rel = .true.)
+
+        end do
+    end subroutine test_logfact_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+    #:for k1, t1 in CI_KINDS_TYPES
+
+    subroutine test_gamma_${t1[0]}$${k1}$(error)
+        type(error_type), allocatable, intent(out) :: error
+        integer, parameter :: n = 4
+        integer :: i
+
+        #:if k1 == "int8"
+
+        ${t1}$, parameter :: x(n) = [1_${k1}$, 2_${k1}$, 4_${k1}$, 6_${k1}$]
+        ${t1}$, parameter :: ans(n) = [1_${k1}$, 1_${k1}$, 6_${k1}$, 120_${k1}$]
+
+        #:elif k1 == "int16"
+
+        ${t1}$, parameter :: x(n) = [1_${k1}$, 2_${k1}$, 4_${k1}$, 8_${k1}$]
+        ${t1}$, parameter :: ans(n) = [1_${k1}$, 1_${k1}$, 6_${k1}$, 5040_${k1}$]
+
+        #:elif k1 == "int32"
+
+        ${t1}$, parameter :: x(n) = [1_${k1}$, 2_${k1}$, 4_${k1}$, 13_${k1}$]
+        ${t1}$, parameter :: ans(n) = [1_${k1}$, 1_${k1}$, 6_${k1}$,           &
+                                      479001600_${k1}$]
+
+        #:elif k1 == "int64"
+
+        ${t1}$, parameter :: x(n) = [1_${k1}$, 2_${k1}$, 4_${k1}$, 21_${k1}$]
+        ${t1}$, parameter :: ans(n) = [1_${k1}$, 1_${k1}$, 6_${k1}$,           &
+                                      2432902008176640000_${k1}$]
+        #:elif t1[0] == "c"
+
+        ${t1}$, parameter :: x(n) = [(0.25_${k1}$, 0.25_${k1}$),               &
+                                     (0.5_${k1}$, -0.5_${k1}$),                &
+                                     (1.0_${k1}$, 1.0_${k1}$),                 &
+                                     (-1.254e1_${k1}$, -9.87_${k1}$)]
+
+        ${t1}$, parameter :: ans(n) =                                          &
+                     [(1.6511332803889208_${k1}$, -1.8378758749947890_${k1}$), &
+                      (0.81816399954174739_${k1}$, 0.76331382871398262_${k1}$),&
+                     (0.49801566811835604_${k1}$, -0.15494982830181069_${k1}$),&
+             (-2.18767396709283064e-21_${k1}$, 2.77577940846953455e-21_${k1}$)]
+        #:endif
+
+
+        #:if t1[0] == "i"
+
+        do i = 1, n
+
+            call check(error, gamma(x(i)), ans(i), "Integer kind ${k1}$ failed")
```

Comment:
Should it be `if(allocated(error)) return` inside the loop after each `check`?

Reply:
not very familiar with test unit. Just followed others' code.